components/org.wso2.carbon.identity.oidc.session/src/test/java/org/wso2/carbon/identity/oidc/session/servlet/OIDCSessionIFrameServletTest.java

```diff
@@ -0,0 +1,146 @@
+
+        initiateInMemoryH2();
+        createOAuthApp(CLIENT_ID_VALUE, SECRET, USERNAME, APP_NAME, "ACTIVE", CALLBACK_URL);
+        createOAuthApp(CLIENT_ID_WITH_NO_CALLBACK_URL, SECRET__WITH_NO_CALLBACK_URL, USERNAME, APP_NAME, "ACTIVE",
+                " ");
+        createOAuthApp(CLIENT_ID_WITH_INVALID_REGEX_CALLBACK_URL, SECRET, USERNAME, APP_NAME, "ACTIVE",
+                INVALID_REGEX_CALLBACK_URL);
+        createOAuthApp(CLIENT_ID_WITH_VALID_REGEX_CALLBACK_URL, SECRET, USERNAME, APP_NAME, "ACTIVE",
+                VALID_REGEX_CALLBACK_URL);
+    }
+
+    /**
+     * This provides data to testDoGet(String clientId, String redirectUri, String expected)
+     * @return
+     */
+    @DataProvider(name = "provideDataForTestDoGet")
+    public Object[][] provideDataForTestDoGet() {
+        return new Object[][]{
+                {CLIENT_ID_VALUE, "", "playground2"},
+                {" ", "", "Invalid"},
+                {INVALID_CLIENT_ID, "", "Invalid"},
+                {CLIENT_ID_WITH_NO_CALLBACK_URL, "", "Invalid"},
+                {CLIENT_ID_WITH_INVALID_REGEX_CALLBACK_URL, "", "Invalid"},
+                {CLIENT_ID_WITH_INVALID_REGEX_CALLBACK_URL, CALLBACK_URL, "Invalid"},
+                {CLIENT_ID_WITH_VALID_REGEX_CALLBACK_URL, CALLBACK_URL, "playground2"}
+        };
+    }
+
+    @Test(dataProvider = "provideDataForTestDoGet")
+    public void testDoGet(String clientId, String redirectUri, String expected) throws Exception {
+        oidcSessionIFrameServlet.init();
+
+        when(request.getParameter("client_id")).thenReturn(clientId);
+        when(request.getParameter(OIDCSessionConstants.OIDC_REDIRECT_URI_PARAM)).thenReturn(redirectUri);
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(oAuthServerConfiguration);
+        when(oAuthServerConfiguration.getPersistenceProcessor()).thenReturn(tokenPersistenceProcessor);
+        when(tokenPersistenceProcessor.getProcessedClientId(anyString())).thenAnswer(new Answer<Object>() {
+
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+
+                return invocation.getArguments()[0];
+            }
+        });
+
+        mockStatic(IdentityTenantUtil.class);
+        when(IdentityTenantUtil.getTenantId(anyString())).thenReturn(TENANT_ID);
+
+        mockStatic(OIDCSessionManagementUtil.class);
+        when(OIDCSessionManagementUtil.getOrigin((CALLBACK_URL))).thenReturn("http://localhost:8080/playground2");
+
+        mockStatic(IdentityDatabaseUtil.class);
+        when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+
+        PrintWriter out = new PrintWriter("oidc-session.txt");
+        when(response.getWriter()).thenReturn(out);
+        oidcSessionIFrameServlet.doGet(request, response);
+        out.flush();
+        assertTrue(FileUtils.readFileToString(new File("oidc-session.txt"), "UTF-8").contains(expected));
```

Comment:
Do  you really need to write this to a file ? . Can't you assert without writing to a file ? . If this is required where are u creating the file ? location ?

Reply:
I can't assert without writing to a file. The test method is a void method and the response is written using a print writer in the test method. That is (response.getWriter().print(ERROR_RESPONSE))